src/couch_db_updater.erl

```diff
@@ -1454,3 +1468,28 @@ default_security_object(_DbName) ->
         "everyone" ->
             []
     end.
+
+% These functions rely on using the process dictionary. This is usually frowned upon
+% howver in this case it is done to avoid changing to a different server state record
+
+schedule_timeout() ->
+    case get(idle_timer_ref) of
+        TimerRef when is_reference(TimerRef) ->
+            erlang:cancel_timer(TimerRef);
+        undefined  ->
+            ok
+    end,
+    put(idle_timer_ref, erlang:send_after(idle_limit(), self(), timeout)).
+
+update_idle_limit_from_config() ->
+    Default = integer_to_list(?IDLE_LIMIT_DEFAULT),
+    IdleLimit = case config:get("couchdb", "idle_check_timeout", Default) of
+        "infinity" ->
+            infinity;
```

Comment:
Well. The only reason we used hibernate was to trigger a full garbage collection. Perhaps we just look at doing that some other way and relying on the timeout. However infinity still means we never pick up the new config parameter as you point out.